crates/nu-command/src/conversions/fmt.rs

```diff
@@ -131,6 +132,40 @@ fn fmt_it(num: i64, span: Span) -> Value {
     Value::Record { cols, vals, span }
 }
 
+fn fmt_it_64(num: f64, span: Span) -> Value {
+    let mut cols = vec![];
+    let mut vals = vec![];
+
+    cols.push("binary".into());
+    vals.push(Value::string(format!("{:b}", num.to_bits()), span));
+
+    cols.push("debug".into());
+    vals.push(Value::string(format!("{num:#?}"), span));
+
+    cols.push("display".into());
+    vals.push(Value::string(format!("{num}"), span));
+
+    cols.push("lowerexp".into());
+    vals.push(Value::string(format!("{num:#e}"), span));
+
+    cols.push("lowerhex".into());
+    vals.push(Value::string(format!("{:x}", num.to_bits()), span));
+
+    cols.push("octal".into());
+    vals.push(Value::string(format!("{:o}", num.trunc() as i64), span));
```

Comment:
I'm wondering if it's more correct to go `to_bits()` versus `trunc()` because at least with `to_bits()` we can undo it like this.
```rust
fn main() {
    let float_number: f32 = 3.14159;

    // Convert to octal
    let bits: u32 = float_number.to_bits();
    let octal_string: String = format!("{:o}", bits);

    // Convert back to float
    let parsed_bits: u32 = u32::from_str_radix(&octal_string, 8).expect("Invalid octal string");
    let converted_float: f32 = f32::from_bits(parsed_bits);

    println!("Original float number: {}", float_number);
    println!("Octal representation: {}", octal_string);
    println!("Converted back: {}", converted_float);
}
```

Reply:
I return it back then

Reply:
event we can use extra parameter for original and bits value if result is most nearest one for float .